src/main/java/ch/njol/skript/effects/EffHealth.java

```diff
@@ -109,14 +111,19 @@ protected void execute(Event event) {
 
 				slot.setItem(itemStack);
 
-			} else if (obj instanceof Damageable) {
-				Damageable damageable = (Damageable) obj;
-
+			} else if (obj instanceof Damageable damageable) {
 				if (this.amount == null) {
 					HealthUtils.heal(damageable, HealthUtils.getMaxHealth(damageable));
 				} else if (isHealing) {
 					HealthUtils.heal(damageable, amount);
 				} else {
+					if (canSetDamageCause && exprCause != null) {
+						DamageCause cause = exprCause.getSingle(event);
+						if (cause != null) {
+							damageable.damage(amount * 2, HealthUtils.getDamageSourceFromCause(cause));
```

Comment:
This isn't consistent with the other methods - you should make a method in HealthUtils for this, which would be mostly the same as `damage(damageable, amount)`, but with an additional `cause` parameter